recipes/darknet/all/conanfile.py

```diff
@@ -43,56 +50,86 @@ def _shared_lib_extension(self):
             return ".so"
 
     def _patch_sources(self):
-        for patch in self.conan_data["patches"].get(self.version, []):
-            tools.patch(**patch)
-        tools.replace_in_file(
-            os.path.join(self._source_subfolder, "Makefile"),
+        apply_conandata_patches(self)
+        replace_in_file(
+            self,
+            os.path.join(self.source_folder, "Makefile"),
             "SLIB=libdarknet.so",
-            "SLIB=libdarknet" + self._shared_lib_extension
+            f"SLIB=libdarknet{self._shared_lib_extension}",
         )
-        tools.replace_in_file(
-            os.path.join(self._source_subfolder, "Makefile"),
+        replace_in_file(
+            self,
+            os.path.join(self.source_folder, "Makefile"),
             "all: obj backup results $(SLIB) $(ALIB) $(EXEC)",
-            "all: obj backup results " + self._lib_to_compile
+            f"all: obj backup results {self._lib_to_compile}",
         )
 
+    def export_sources(self):
+        export_conandata_patches(self)
+
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
 
-    def validate(self):
-        if self.settings.os == "Windows":
-            raise ConanInvalidConfiguration("This library is not compatible with Windows")
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def requirements(self):
         if self.options.with_opencv:
+            # Requires OpenCV 2.x
             self.requires("opencv/2.4.13.7")
 
+    def validate(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("This library is not compatible with Windows")
+
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder,
-                  strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = AutotoolsToolchain(self)
+        tc.make_args = ["OPENCV={}".format("1" if self.options.with_opencv else "0")]
+        tc.generate()
+        tc = PkgConfigDeps(self)
+        tc.generate()
 
     def build(self):
-        self._patch_sources()
-        with tools.chdir(self._source_subfolder):
-            with tools.environment_append({"PKG_CONFIG_PATH": self.build_folder}):
-                args = ["OPENCV={}".format("1" if self.options.with_opencv else "0")]
-                env_build = AutoToolsBuildEnvironment(self)
-                env_build.fpic = self.options.get_safe("fPIC", True)
```

Comment:
The builds are failing for ``shared=True``, because it is missing the ``-fPIC`` flag, that seems explicitly necessary in this project. the ``AutotoolsToolchain`` has a ``self.fpic = self._conanfile.options.get_safe("fPIC")`` that can be also overriden by default in the same way.

Reply:
And thanks for your contribution, you are doing an amazing job migrating recipes to Conan 2.0!